public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -362,10 +388,132 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
           )}
         </Stack>
       )}
+      <SmartAlertTypeDetector
+        editingExistingRule={editingExistingRule}
+        rulesSourcesWithRuler={rulesSourcesWithRuler}
+        queries={queries}
+      />
```

Comment:
I would only show this component if not editing and not a recording rule, meaning I'd remove the `editingExistingRule` prop and the responsibility to know whether to show itself or not from within the `SmartAlertTypeDetector` and move the condition outside of it like:



```suggestion
      {!editingExistingRule && !isRecordingRuleType && (
        <SmartAlertTypeDetector rulesSourcesWithRuler={rulesSourcesWithRuler} queries={queries} />
      )}
```

Reply:
We cannot do that as we want to show the banner when editing the alert rule (without the switch).